lib/super_digit.rb

```diff
@@ -1,15 +1,33 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n) n being number of digits
+# Space Complexity - O(n)
 def super_digit(n)
```

Comment:
Note that your loop is O(n) and it's nested inside recursion.  So the time/space complexity is worse than O(n).